src/Sylius/Behat/Context/Setup/PromotionContext.php

```diff
@@ -564,6 +565,22 @@ public function thePromotionGivesOffIfOrderContainsProducts(PromotionInterface $
     }
 
     /**
+     * @Given /^([^"]+) gives ("(?:€|£|\$)[^"]+") off on a ("[^"]*" product)$/
+     */
+    public function itGivesFixedDiscountOffOnAProduct(PromotionInterface $promotion, $discount, ProductInterface $product)
+    {
+        $this->createUnitFixedPromotion($promotion, $discount, $this->getProductsFilterConfiguration([$product->getCode()]));
+    }
+
+    /**
+     * @Given /^([^"]+) gives ("[^"]+%") off on a ("[^"]*" product)$/
+     */
+    public function itGivesPercentageDiscountOffOnAProduct(PromotionInterface $promotion, $discount, ProductInterface $product)
+    {
+        $this->createUnitPercentagePromotion($promotion, $discount, $this->getProductsFilterConfiguration([$product->getCode()]));
```

Comment:
But thats a point of this test. This promotion should be assigned to every order, but not every item should be affected. The following scenario checks if filter works properly:
```yml
    Background:
        Given the store operates on a single channel in "United States"
        And the store has a product "PHP T-Shirt" priced at "$100.00"
        And the store has a product "PHP Mug" priced at "$20.00"
        And there is a promotion "T-Shirts promotion"
        And it gives "$10.00" off on a "PHP T-Shirt" product

    @ui
    Scenario: Receiving fixed discount on a single item
        When I add product "PHP T-Shirt" to the cart
        Then its price should be decreased by "$10.00"
        And my cart total should be "$90.00"

    @ui
    Scenario: Not receiving fixed discount on another item
        When I add product "PHP Mug" to the cart
        Then product "PHP Mug" price should not be decreased
        And my cart total should be "$20.00"
```